lib/atecc508a/request.ex

```diff
@@ -196,6 +195,46 @@ defmodule ATECC508A.Request do
   defp length_flag(32), do: 1
   defp length_flag(4), do: 0
 
+  @spec transport_request(
+          transport :: Transport.t(),
+          payload :: binary(),
+          timeout :: non_neg_integer(),
+          response_payload_len :: non_neg_integer(),
+          request_timeout :: non_neg_integer()
+        ) :: {:ok, binary()} | {:error, atom()}
+  defp transport_request(transport, payload, timeout, response_payload_len, retry \\ 1000) do
+    now = System.monotonic_time(:second)
```

Comment:
```suggestion
    give_up_time = System.monotonic_time(:millisecond) + timeout
```
I think this should be milliseconds since the default is 1000. I also changed to `give_up_time` since that seemed to make the recursion simpler below.